From 524a01a08d6f253f01a3e7dd6c35a66fe21f8c30 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Mon, 28 Apr 2014 10:51:36 +0800
Subject: [PATCH 1/3] check lokkit before run it

Check metacity and lokkit before run it to avoid such errors in /tmp/program.log

02:06:31,146 ERR anaconda: Error running /usr/sbin/lokkit: No such file or directory
02:06:31,158 ERR anaconda: lokkit run failed: Error running /usr/sbin/lokkit: No such file or directory
02:06:37,830 ERR anaconda: Error running /usr/sbin/authconfig: No such file or directory
02:06:37,844 ERR anaconda: Error running ['--update', '--nostart', '--enableshadow', '--passalgo=sha512']: Error running /usr/sbin/authconfig: No such file or directory
02:06:37,917 ERR anaconda: Error running /usr/sbin/lokkit: No such file or directory
02:06:37,929 ERR anaconda: lokkit run failed: Error running /usr/sbin/lokkit: No such file or directory

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/firewall.py | 4 ++++
 pyanaconda/security.py | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/pyanaconda/firewall.py b/pyanaconda/firewall.py
index c11b671..4cde8ea 100644
--- a/pyanaconda/firewall.py
+++ b/pyanaconda/firewall.py
@@ -71,6 +71,10 @@ class Firewall:
 	return args
 
     def write (self, instPath):
+        if not os.path.exists("/usr/sbin/lokkit"):
+            log.info("Skipping running /usr/sbin/lokkit since it doesn't exist.")
+            return 0
+
 	args = [ "--quiet", "--nostart", "-f" ] + self.getArgList()
 
         try:
diff --git a/pyanaconda/security.py b/pyanaconda/security.py
index 6f9449b..3dbf359 100644
--- a/pyanaconda/security.py
+++ b/pyanaconda/security.py
@@ -68,6 +68,11 @@ class Security:
         return ts.dbMatch('provides', 'fprintd-pam').count()
 
     def write(self, instPath):
+        import os
+        if not os.path.exists("/usr/sbin/lokkit"):
+            log.info("Skipping running /usr/sbin/lokkit since it doesn't exist.")
+            return 0
+
         args = []
 
         if not selinux_states.has_key(self.selinux):
-- 
1.8.3.1

